MERN/Topic/Backend/Node/4.1_How to use asynchronous APIs.md

```diff
@@ -0,0 +1,302 @@
+## What is an API?
+
+APIs are application programming interfaces that return data on request immediately or after sometime depending upon the code.
+
+API can be either synchronous or asynchronous.
+
+- Synchronous returns the data immediately,wherein the data is requested and waits for it until a value is returned ,it executes from top to bottom of a file or application one by one.
+
+- Aynchronous APIs provide a callback to the requester when the requested resource/value is ready,rather than waiting on a request.
+
+*It may or may not execute synchronously*
+
+*When it encounters asynchronous functions or code ,it splits off while executing that asynchronous function separately along with the rest of the code,therefore it may happen that it does not execute in a synchronous manner.(One may execute faster than the other)*
```

Comment:
Please try to explain this in a little more detailed and easy way For example instead of splitting off you can say that being asynchronous it keeps running the code after it and does not wait for the request to complete.